### PR TITLE
boards: xtensa: m5stickc_plus: initial support

### PR DESCRIPTION
**Pull request by JZimnol**
Adds initial support for `M5StickC-Plus` board.

Adding a new board was strongly based on the `ESP-WROVER-KIT` board example. For not (yet) supported sensors, `*.dts` files has been prepared for adding them e.g. by setting proper pins for `i2c0` or `spi3`. 

Despite limited board pinout, I've added [openocd.cfg](https://github.com/zephyrproject-rtos/zephyr/compare/main...JZimnol:zephyr:bsp/m5stickc_plus?expand=1#diff-52b5f8e75410e30f855fb3174acd818b7474efded44eaf6c9777816546a92d31) file in case user wants to connect external debugger directly to the internal chip (an unlikely case, but still possible).

Signed-off-by: Jakub Zimnol <j.zimnol@avsystem.com>

### CODE REVIEW

File: boards/xtensa/m5stickc_plus/m5stickc_plus.dts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+// IMU MPU-6886, RTC BM8563, PMU AXP192
```
Comment (dcpleung) [inline boards/xtensa/m5stickc_plus/m5stickc_plus.dts:76]: Please use `/* */` style comments.

File: boards/xtensa/m5stickc_plus/m5stickc_plus.dts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+&spi2 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+	pinctrl-0 = <&spim2_miso_gpio12 &spim2_mosi_gpio11
+				&spim2_sclk_gpio14 &spim2_csel_gpio16>;
+	pinctrl-names = "default";
+};
+
+// LCD TFT 1.14", 135x240 px, ST7789v2
```
Comment (dcpleung) [inline boards/xtensa/m5stickc_plus/m5stickc_plus.dts:95]: Ditto

**Review by dcpleung — CHANGES_REQUESTED** [review]
Please rebase to get rid of the merge commit.

**Comment by dcpleung** [comment]
Please squash the two commits together.

**Comment by JZimnol** [comment]
@dcpleung another round of review requested.

File: boards/xtensa/m5stickc_plus/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+|                  | low-power consumption while the chip is in Deep-sleep mode.             |
++------------------+-------------------------------------------------------------------------+
+| ESP32-PICO-D4    | This `ESP32-PICO-D4`_ module provides complete Wi-Fi and Bluetooth      |
+| module           | functionalities and integrates a 4-MB SPI flash.                        |
++------------------+-------------------------------------------------------------------------+
+| Diagnostic LED   | One user LED connected to the GPIO pin.                                 |
++------------------+-------------------------------------------------------------------------+
+| USB Port         | USB interface. Power supply for the board as well as the                |
+|                  | communication interface between a computer and the board.               |
+|                  | Contains: TypeC x 1, GROVE(I2C+I/0+UART) x 1                            |
```
Comment (dcpleung) [inline boards/xtensa/m5stickc_plus/doc/index.rst:47]: Maybe typo? `I/O` instead of `I/0` (letter 'O' instead of zero)?

Comment (JZimnol) [reply boards/xtensa/m5stickc_plus/doc/index.rst]: @dcpleung it was indeed a typo, it was copied from one of the board's documentation files (now I've double checked). Changes applied.

**Review by dcpleung — DISMISSED** [review]

**Comment by JZimnol** [comment]
@dcpleung I've updated initial files so the board passes tests with the newest SDK. The only failing test is the `zephyr/samples/arch/smp/pktqueue` because of the memory overflow, but none of the `esp32` boards passes this test (tested manually).

**Review by dcpleung — DISMISSED** [review]

**Review by Mierunski — DISMISSED** [review]

**Comment by dcpleung** [comment]
If that test never passes, this board needs to be added to the `platform_exclude` list (similar to other esp32 boards).

**Review by dcpleung — DISMISSED** [review]

**Comment by sylvioalves** [comment]
@JZimnol There is still CI issues and file conflict. I'll also review the PR later. I wasn't add before, sorry for missing this.

**Comment by JZimnol** [comment]
@sylvioalves:
> @JZimnol There is still CI issues and file conflict. I'll also review the PR later. I wasn't add before, sorry for missing this.

Yeah, I had some trouble with conflicts, because my forked `main branch` was quite out of date. Everything is OK now and can be reviewed.

**Comment by JZimnol** [comment]
And, again, I re-requested review from @dcpleung and GitHub automatically removed request from @nashif and @sylvioalves.

File: boards/xtensa/m5stickc_plus/m5stickc_plus_defconfig [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_XTENSA_RESET_VECTOR=n
+
+CONFIG_BOARD_M5STICKC_PLUS=y
+CONFIG_SOC_ESP32=y
+
+CONFIG_MAIN_STACK_SIZE=2048
+
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=240000000
```
Comment (sylvioalves) [inline boards/xtensa/m5stickc_plus/m5stickc_plus_defconfig:10]: You could remove this line as `SYS_CLOCK_HW_CYCLES_PER_SEC` already is set to default to .DTS 240MHz.

File: boards/xtensa/m5stickc_plus/m5stickc_plus_defconfig [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+
+CONFIG_XTENSA_USE_CORE_CRT1=n
+
+CONFIG_GPIO=y
+CONFIG_GPIO_ESP32=y
```
Comment (sylvioalves) [inline boards/xtensa/m5stickc_plus/m5stickc_plus_defconfig:19]: No need for `CONFIG_GPIO_ESP32=y` anymore as it is automatically set.

**Comment by sylvioalves** [comment]
@JZimnol, LGTM. Just those 2 comments regarding defconfig.

File: boards/xtensa/m5stickc_plus/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,137 @@
+.. _m5stickc_plus:
+
+M5StickC PLUS
+##############
```
Comment (sylvioalves) [inline boards/xtensa/m5stickc_plus/doc/index.rst:4]: There is an extra `#` in there.

File: boards/xtensa/m5stickc_plus/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+Prerequisites
+-------------
+
+Espressif HAL requires binary blobs in order work. The west extension below performs the required
+synchronization to clone, checkout and pull the submodules:
+
+.. code-block:: console
+
+   west espressif update
```
Comment (sylvioalves) [inline boards/xtensa/m5stickc_plus/doc/index.rst:81]: Would you update this documentation? This command is no longer available. Please, check `esp32` board documentation as reference.

**Comment by sylvioalves** [comment]
@JZimnol, sorry I haven't spotted those items I commented on previous review.

**Comment by sylvioalves** [comment]
Fix your commit as well, it is missing the `Signed-off` line.

**Comment by JZimnol** [comment]
@sylvioalves thank you for your comments! Those commands were indeed outdated, because I created this MR using SDK v0.14.1 or something. I hope everything is OK now.

**Review by sylvioalves — APPROVED** [review]

File: boards/xtensa/m5stickc_plus/m5stickc_plus.dts [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+	status = "okay";
+};
+
+/* IMU MPU-6886, RTC BM8563, PMU AXP192 */
+&i2c0 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-gpios = <&gpio0 21 GPIO_OPEN_DRAIN>;
+	scl-gpios = <&gpio0 22 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c0_sda_gpio21 &i2c0_scl_gpio22>;
```
Comment (marekmatej) [inline boards/xtensa/m5stickc_plus/m5stickc_plus.dts:83]: you could also use `&i2c0_default` from `esp32-pinctrl.dtsi` as it is using same pins

**Review by marekmatej — APPROVED** [review]

**Pull request merged**